core/src/main/java/bio/terra/pearl/core/service/export/integration/ExternalExporter.java

```diff
@@ -0,0 +1,65 @@
+package bio.terra.pearl.core.service.export.integration;
+
+import bio.terra.pearl.core.model.export.ExportIntegration;
+import bio.terra.pearl.core.model.export.ExportIntegrationJob;
+import bio.terra.pearl.core.service.export.EnrolleeExportService;
+import bio.terra.pearl.core.service.export.ExportOptionsWithExpression;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.scheduling.annotation.Async;
+
+import java.io.ByteArrayOutputStream;
+import java.util.function.Consumer;
+
+@Slf4j
+public abstract class ExternalExporter {
+
+    private final ExportIntegrationJobService exportIntegrationJobService;
+
+    private final EnrolleeExportService enrolleeExportService;
+
+    protected abstract void send(ExportIntegration integration, ByteArrayOutputStream baos, Consumer<String> handleComplete, Consumer<Exception> handleError);
```

Comment:
```suggestion
    protected abstract void send(ExportIntegration integration, ByteArrayOutputStream exportStream, Consumer<String> handleComplete, Consumer<Exception> handleError);
```
(just for clarity, took a me a sec to figure out what baos was)

Reply:
it was lunch time when I coded this :)